### PR TITLE
JIT: ARM64 SVE format encodings, `SVE_BF_2A` to `SVE_FU_2A`

### PR DESCRIPTION
**Pull request by TIHan**
Contributes to https://github.com/dotnet/runtime/issues/94549

Adds formats:
- SVE_BF_2A
- SVE_FT_2A
- SVE_FU_2A

These were tricky formats as the immediate is encoded as part of the element size. I added pretty exhaustive tests to ensure that this is correct.

Left: Capstone
Right: Jit
image
image
image

### CODE REVIEW

**Comment by TIHan** [comment]
@dotnet/jit-contrib @dotnet/arm64-contrib @a74nh @amanasifkhalid this is ready.

File: src/coreclr/jit/emitarm64.cpp [hunk 1]
```diff
@@ -18577,6 +18629,69 @@ void emitter::emitIns_Call(EmitCallType          callType,
     return encoding;
 }
 
+/*****************************************************************************
+ *
+ *  Returns the encoding for the field 'tszh:tszl:imm3' at bit locations '23-22:20-19:18-16'.
+ */
+
+/*static*/ emitter::code_t emitter::insEncodeSveElemsizeWithShift_tszh_tszl_imm3(const insOpts opt,
+                                                                                 ssize_t       imm,
```
Comment (a74nh) [inline src/coreclr/jit/emitarm64.cpp:18638]: Shift immediates are notoriously awkward :)

Does any of this code overlap with the existing `insEncodeShiftImmediate()`?
You can see how it's used in `insEncodeSveShift_23_to_22_9_to_0()`
`insEncodeShiftImmediate()` is also used by NEON, as shift immediates tend to be fairly standard.

**Review by amanasifkhalid — APPROVED** [review]
I second Alan's comment on using `insEncodeShiftImmediate` in `insEncodeSveElemsizeWithShift_tszh_tszl_imm3`. Everything else LGTM, thanks!

File: src/coreclr/jit/emitarm64.cpp (hunk 1 above)
Comment (TIHan) [reply src/coreclr/jit/emitarm64.cpp:18638]: I might be able to use `insEncodeShiftImmediate` to tweak the `imm` instead of doing it manually.

**Pull request merged**